doc/dev/deduplication.rst

```diff
@@ -0,0 +1,253 @@
+and its reference count information.
+
+
+Although chunk objects and manifest objects have a different purpose 
+from existing objects, they can be handled the same way as 
+original objects. Therefore, to support existing features such as replication,
+no additional operations for dedup are needed.
+
+Usage
+=====
+
+To set up deduplication pools, you must have two pools. One will act as the 
+base pool and the other will act as the chunk pool. The base pool need to be
+configured with fingerprint_algorithm option as follows.
+
+::
+
+  ceph osd pool set $BASE_POOL fingerprint_algorithm sha1|sha256|sha512 
+  --yes-i-really-mean-it
+
+1. Create objects ::
+
+        - rados -p base_pool put foo ./foo
+
+        - rados -p chunk_pool put foo-chunk ./foo-chunk
+
+2. Make a manifest object ::
+
+        - rados -p base_pool set-chunk foo $START_OFFSET $END_OFFSET --target-pool 
+        chunk_pool foo-chunk $START_OFFSET --with-reference
+
+
+Interface
+=========
+
+* set-redirect 
+
+  set redirection between a base_object in the base_pool and a target_object 
+  in the target_pool.
+  A redirected object will forward all operations from the client to the 
+  target_object. ::
+  
+        rados -p base_pool set-redirect <base_object> --target-pool <target_pool> 
+         <target_object>
+
+* set-chunk 
+
+  set chunk-offset in a source_object to make a link between it and a 
+  target_object. ::
+  
+        rados -p base_pool set-chunk <source_object> <offset> <length> --target-pool 
+         <caspool> <target_object> <taget-offset> 
+
+* tier-promote 
+
+  promote the object (including chunks). ::
+
+        rados -p base_pool tier-promote <obj-name> 
+
+* unset-manifest
+
+  unset manifest optioan form the object that has manifest. ::
+
+        rados -p base_pool unset-manifest <obj-name>
+
+
+Dedup tool
+==========
+
+Dedup tool has two features: finding optimal chunk offset for dedup chunking 
+and fixing the reference count.
+
+* find optimal chunk offset
+
+  a. fixex chunk  
+
+    To find out a fixed chunk length, you need to run following command many 
+    times while changing the chunk_size. ::
+
+            ceph-dedup-tool --op estimate --pool $POOL --chunk-size chunk_size  
+              --chunk-algorithm fixed --fingerprint-algorithm sha1|sha256|sha512
+
+  b. rabin chunk(Rabin-karp algorithm) 
+
+    As you know, Rabin-karp algorithm is string-searching algorithm based
+    on a rolling-hash. But rolling-hash is not enough to do deduplication because 
+    we don't know the chunk boundary. So, we need content-based slicing using 
+    a rolling hash for content-defined chunking.
+    The current implementation uses the simplest approach: look for chunk boundaries 
+    by inspecting the rolling hash for pattern(like the
+    lower N bits are all zeroes). 
+      
+    - Fixed chunk vs content-defined chunk
+
+      Content-defined chunking may or not be optimal solution.
+      For example,
```

Comment:
the format of this part is messed up after rendered by sphinx. i guess you could use newlines to separate the paragraphs ?

JFYI, you could follow the instructions at http://docs.ceph.com/docs/master/start/documenting-ceph/#build-the-source to build the doc and check it locally. so you can improve this document with faster iterations.